pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -426,7 +426,7 @@ func validateImmutableFieldsCluster(new, old *Cluster) field.ErrorList {
 		}
 	}
 
-	if !oldAWSIamConfig.Equal(newAWSIamConfig) {
+	if !oldAWSIamConfig.IsEmpty() && !newAWSIamConfig.IsEmpty() && !oldAWSIamConfig.Equal(newAWSIamConfig) {
```

Comment:
What is the intention here? Like if the cluster had an IAMconfig and a new one is coming they have to be the same?
Isnt that the same as `!oldAWSIamConfig.Equal(newAWSIamConfig)`? or am i missing something?

Reply:
If old or new ref is empty, we will not enforce any validations going forward. This will disable validations in the cases of addition or deletion of aws iam authentication to an existing cluster.

Reply:
So we are blocking only modification and not addition deletion?